modules/services/vegbilder.js

```diff
@@ -0,0 +1,609 @@
+        const b = projection(lastImage.loc);
+        const a = projection(image.loc);
+        if (!geoVecEqual(a, b)) {
+          angle = geoVecAngle(a, b);
+          angle *= (180 / Math.PI);
+          angle -= 90;
+          angle = angle >= 0 ? angle : angle + 360;
+        }
+        lastImage.ca = angle;
+      }
+      if (
+              image.direction === lastImage.direction &&
+              image.captured_at.valueOf() - lastImage.captured_at.valueOf() <= 20000
+              ) {
+        imageSequence.push(image);
+      } else {
+        imageSequences.push(imageSequence);
+        imageSequence = [image];
+      }
+    }
+    imageSequences.push(imageSequence);
+    return imageSequences;
+  }, []);
+
+  cache.sequences = imageSequences.map(images => {
+    const seqence = {
+      images,
+      key: images[0].key,
+      geometry : {
+        type : 'LineString',
+        coordinates : images.map(image => image.loc)
+      }};
+    for (const image of images) {
+      _vegbilderCache.image2sequence_map.set(image.key, seqence);
+    }
+    return seqence;
+  });
+}
+
+function roadReference(properties) {
+  const {
+    FYLKENUMMER: county_number,
+    VEGKATEGORI: road_class,
+    VEGSTATUS: road_status,
+    VEGNUMMER: road_number,
+    STREKNING: section,
+    DELSTREKNING: subsection,
+    HP: parcel,
+    KRYSSDEL: junction_part,
+    SIDEANLEGGSDEL: services_part,
+    ANKERPUNKT: anker_point,
+    AAR: year,
+  } = properties;
+
+  let reference;
+
+  if (year >= 2020) {
+    reference = `${road_class}${road_status}${road_number} S${section}D${subsection}`;
+    if (junction_part) {
+      reference = `${reference} M${anker_point} KD${junction_part}`;
+    } else if (services_part) {
+      reference = `${reference} M${anker_point} SD${services_part}`;
+    }
+  } else {
+    reference = `${county_number}${road_class}${road_status}${road_number} HP${parcel}`;
+  }
+
+  return reference;
+}
+
+function localeTimestamp(date) {
+  const options = { day: '2-digit', month: '2-digit', year: 'numeric',
+    hour: 'numeric', minute: 'numeric', second: 'numeric' };
+  return date.toLocaleString(localizer.localeCode(), options);
+}
+
+function partitionViewport(projection) {
+  const zoom = geoScaleToZoom(projection.scale());
+  const roundZoom = (Math.ceil(zoom * 2) / 2) + 2.5;   // round to next 0.5 and add 2.5
+  const tiler = utilTiler().zoomExtent([roundZoom, roundZoom]);
+
+  return tiler.getTiles(projection)
+    .map(tile => tile.extent);
+}
+
+function searchLimited(limit, projection, rtree) {
+  limit ??= 5;
+
+  return partitionViewport(projection)
+    .reduce((result, extent) => {
+      const found = rtree.search(extent.bbox())
+        .slice(0, limit)
+        .map(d => d.data);
+
+      return result.concat(found);
+    }, []);
+}
+
+
+export default {
+
+  init: async function () {
+    if (!_vegbilderCache) {
+      await this.reset();
+    }
+
+    this.event = utilRebind(this, dispatch, 'on');
+  },
```

Comment:
I've noticed that this can cause an exception in [`renderer/photos.js`](https://github.com/openstreetmap/iD/blob/ddc9c6c67460e20a9531e6c759c61bebc5269b80/modules/renderer/photos.js#L209) (when opening a photo via URL parameter is requested).

I think we should not `await` the cache reset before binding the `on` method here:

 ```suggestion
  init: async function () {
    this.event = utilRebind(this, dispatch, 'on');
    if (!_vegbilderCache) {
      await this.reset();
    }
  },
```

But then `loadImages` needs to check if the cache is already populated:

```javascript
  loadImages: async function (context, margin) {
    margin ??= 1;
    if (!_vegbilderCache) {
      await this.reset();
    }
    const wfslayers = filterAvailableLayers(context.photos());
```

The same also in the `sequences` method plus some added async/await on top.

<hr>

Maybe there's a better option to fix this.

Reply:
I have rearranged `init` as you suggested, and rearranged  `reset` to initialize a empty cache right away, before waiting for `fetchAvailableLayers`. Then it should not be necessary to check if the cache is there, in every function accessing the cache. Is that a better solution?